### ISSUE TITLE
CI build is broken

### ISSUE THREAD

**Issue by kazimuth** [Tiramisu-Compiler/tiramisu#297]
The CI build [is failing](https://travis-ci.org/Tiramisu-Compiler/tiramisu/builds/548327337#L4815). It seems to have something to do with halide's ABI not matching LLVM's, not sure how to fix that.

**Comment by rbaghdadi** [comment]
Hi @kazimuth , yes indeed. If we use the new Halide (and not our fork of Halide which is a bit old), that would solve the problem. On one of our internal machines, Tiramisu works fine with a new Halide, but I just did not have time to propagate the change to CI.

If you are interested in helping that I can give you assistance and we can try.

**Comment by kazimuth** [comment]
I might be able to pick that up, sure. I'm contemplating using Tiramisu for some MIT coursework this semester, but to do that I'll need a functioning build on my machine :)

**Comment by rbaghdadi** [comment]
Oh Ok I see. That one is independent though. You should be able to compile and use Tiramisu without a problem. The problem that we have with CI is only related to a mismatch in the LLVM versions we are using or something. There is a PR that tries to address that, but I still didn't have time to test it.

Go ahead and just install Tiramisu on your machine and let us know if you have a problem. If you get a problem I'll be happy to help you out.

**Comment by kazimuth** [comment]
Yeah, I was running into the same problem cause I was trying to avoid recompiling LLVM lol.

Will do.

**Comment by rbaghdadi** [comment]
Ok, here is the solution then:
- Compile a fresh Halide as described in the Halide github page: https://github.com/halide/Halide (you do not need to compiler LLVM from sources)
- Then use that Halide to compile Tiramisu. You need only to edit the two lines here: https://github.com/Tiramisu-Compiler/tiramisu/blob/8ca4173547b6d12cff10575ef0dc48cf93f7f414/configure.cmake#L49 
to point to your new Halide.

In principle, if you get a fresh Halide from the halide repo, that one is good. The Halide that we have is a bit old and does not have the fix to avoid the problem you specified.

**Comment by rbaghdadi** [comment]
Let me know if this works for you.

**Comment by kazimuth** [comment]
Compiling old LLVM and halide works fine.

Using current halide gives a few compile errors; they seem fixable, though. I might take a shot at patching them over the weekend.

**Comment by rbaghdadi** [comment]
Ok this is perfect! So if I understood well, you problem is solved now?

You want to patch the old Halide or the new one?

**Comment by kazimuth** [comment]
I think it would be possible to just patch *tiramisu* to use the new halide. Most of the changes seem to be small changes to function signatures, unless there have been any major underlying architecture changes in Halide I'm not seeing.

**Comment by rbaghdadi** [comment]
In theory, even Tiramisu should not be modified. It should work with the new Halide without a change. Only the github submodule pointer has to point to the public halide instead of pointing to the current outdated Halide. I have tried before to make it work with the new Halide and it just worked with the need to change Tiramisu so it should be fine!  (unless I'm missing something)

**Comment by kazimuth** [comment]
Hm, there seem to be some compilation errors? Changed signatures and stuff, with the most recent halide release.

**Comment by rbaghdadi** [comment]
Sorry which errors are you referring to?

**Comment by mratsim** [comment]
Are there new errors compared to those I reported? https://github.com/Tiramisu-Compiler/tiramisu/issues/217 (note that I didn't try to build Tiramisu with Halide master)
